frontend/src/pages/chat/Chat.tsx

```diff
@@ -32,26 +34,64 @@ const Chat = () => {
         lastQuestionRef.current = question;
 
         setIsLoading(true);
+        setShowLoadingMessage(true);
         const abortController = new AbortController();
         abortFuncs.current.unshift(abortController);
 
         const userMessage: ChatMessage = {
             role: "user",
-            content: question,
-            end_turn: null
+            content: question
         };
 
         const request: ConversationRequest = {
             messages: [...answers, userMessage]
         };
 
+        let result = {} as ChatResponse;
         try {
-            const result = await conversationApi(request, abortController.signal);
-            setAnswers([...answers, userMessage, ...result.choices[0].messages]);
-        } catch {
+            const response = await conversationApi(request, abortController.signal);
+            if (response?.body) {
+                
+                const reader = response.body.getReader();
+                let runningText = "";
+                while (true) {
+                    const {done, value} = await reader.read();
+                    if (done) break;
+
+                    var text = new TextDecoder("utf-8").decode(value);
+                    console.log(text);
+                    const objects = text.split("}{");
```

Comment:
I think this would break if the reply happened to contain `}{`.

Reply:
May want to look at splitting on newlines and having app.py automatically add a \n after doing json.dumps. app.py could also replace any newlines in the text with \\n in case they occur in the text (or that may happen already, not sure).  By splitting on newlines, you can remove 65-70 below.

Reply:
Added a "<newline>" for a safer split text